### PR TITLE
bugfix/BB 366/addGCReadAccountIDs

### PR DESCRIPTION
**Pull request by alexanderchan-scality**
- BB-366: add gc read_accounts policies
- BB-366: GC task - add accountId get for replication generated entries
- fix linter warnings

### CODE REVIEW

File: extensions/gc/GarbageCollector.js [hunk 1]
```diff
@@ -10,6 +10,12 @@ const ActionQueueEntry = require('../../lib/models/ActionQueueEntry');
 const ClientManager = require('../../lib/clients/ClientManager');
 const GarbageCollectorTask = require('./tasks/GarbageCollectorTask');
 
+const { authTypeAssumeRole } = require('../../lib/constants');
+
+// const { AccountIdCache } = require('../utils/AccountIdCache');
```
Comment (nicolas2bert) [inline extensions/gc/GarbageCollector.js]: to be removed.

File: extensions/gc/tasks/GarbageCollectorTask.js [hunk 2]
```diff
@@ -19,87 +19,119 @@ class GarbageCollectorTask extends BackbeatTask {
...
-    _getMetadata(entry, log, done) {
-        const { accountId } = entry.getAttribute('target');
-        const backbeatClient = this.getBackbeatMetadataProxy(accountId);
-        if (!backbeatClient) {
-            log.error('failed to get backbeat client', { accountId });
-            return done(errors.InternalError
-                .customizeDescription('Unable to obtain client'));
+    // helper method needed for replication generated entries in which the
+    // account id information is missing.
+    // TODO: replace once replication services uses assume role
```
Comment (nicolas2bert) [inline extensions/gc/tasks/GarbageCollectorTask.js]: Is there a ticket for it?
Please mention it.

File: extensions/gc/tasks/GarbageCollectorTask.js [hunk 3]
```diff
@@ -19,87 +19,119 @@ class GarbageCollectorTask extends BackbeatTask {
...
-            versionId: version,
-        }, log, (err, blob) => {
+        log.debug('unable to find account id in entry; performing vault request', {
+            ownerId,
+        });
+        return this.getAccountId(ownerId, log, cb);
+    }
+
+    _getMetadata(entry, log, done) {
+        this._getAccountId(entry, log, (err, accountId) => {
```
Comment (nicolas2bert) [inline extensions/gc/tasks/GarbageCollectorTask.js:39]: Do we cache the account id?

**Review by naren-rajendran — APPROVED** [review]

File: policies/read_accounts.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+    "Version": "2012-10-17",
+    "Statement": [{
+        "Sid": "ReadAccounts",
+        "Effect": "Allow",
+        "Action": [
+            "vaultadmin:GetAccountInfo"
```
Comment (ghost) [inline policies/read_accounts.json:7]: Do we support this action for non root users today? I don't see it in the actionMaps...

**Review by ghost — APPROVED** [review]

**Review by nicolas2bert — APPROVED** [review]

**Comment by alexanderchan-scality** [comment]
ping

**Pull request merged**